library/ssl_tls.c

```diff
@@ -175,7 +175,7 @@ int mbedtls_ssl_check_record( mbedtls_ssl_context const *ssl,
                               unsigned char *buf,
                               size_t buflen )
 {
-    int ret = 0;
+    int ret = MBEDTLS_ERR_SSL_INTERNAL_ERROR;
```

Comment:
Do I see well that for `MBEDTLS_SSL_PROTO_TLS` and without` MBEDTLS_SSL_PROTO_DTLS` this function will never return success?

Reply:
Yes, as records are not used for TLS. See comments at line 183.